client/src/components/TopNavigationBar.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+import {
+  AppBar,
+  Typography,
+  Toolbar,
+  Button,
+  Menu,
+  MenuItem,
+  Divider,
+} from "@material-ui/core";
+import { ArrowDropDown } from "@material-ui/icons";
+import { makeStyles } from "@material-ui/core/styles";
+import { ReactComponent as Logo } from "images/logo.svg";
+import UserAvatar from "common/UserAvatar";
+
+const TopNavigationBar = ({ loggedInUser }) => {
+  const classes = useStyles();
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
+  const handleMenu = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  return (
+    <AppBar style={{ background: "#fff" }} position="static">
+      <Toolbar>
+        <Logo className={classes.logo} />
```

Comment:
Logo and Profile links in navigation should link to their respective routes, `<Link href="/" color="inherit">
          <Logo className={classes.logo} />
        </Link>`
From reading [Material UI doc for Link](https://material-ui.com/components/links/), code should be above. but when i tried doing so on my local branch clicking the links would set global user to `null` and log me out. Either I set up the link wrong, or somewhere in code is a bug that is preventing global user state from persisting long term in context.

Reply:
I misunderstood Material UI links. Turns out they are just for styling purposes. To make it work you'll have to combine them with a Link from react-router-dom as shown here in example of [with prop forwarding](https://material-ui.com/guides/composition/#link). Once thats done the link will navigate correctly and will keep user logged in.